### PR TITLE
[CI][HIP] Add missing dep for broken amd script.

### PR DESCRIPTION
**Pull request by JackAKirk**
Fixes https://github.com/intel/llvm/issues/13612

https://github.com/intel/llvm/issues/13612 was caused by these issues:

https://github.com/ROCm/ROCm/issues/2949
https://github.com/ROCm/HIP/issues/3406

To fix this gcc/g++12 is installed manually and set as the preferred version.

### CODE REVIEW

File: devops/containers/ubuntu2204_build.Dockerfile [hunk 1]
```diff
@@ -7,18 +7,27 @@ USER root
...
   apt install -yqq ./amdgpu-install_6.1.60100-1_all.deb && \
-  yes | amdgpu-install --usecase=rocmdev && \
+  amdgpu-install -y --usecase=rocmdev && \
   apt-get clean && \
   rm -rf /var/lib/apt/lists/*
 
+RUN g++ --version && \
+  gcc --version
+RUN ln -s -f /usr/bin/gcc-12 /usr/bin/gcc && \
+  ln -s -f /usr/bin/g++-12 /usr/bin/g++
```
[the remark this answers is not in the document]
Comment (JackAKirk) [reply devops/containers/ubuntu2204_build.Dockerfile]: Thanks, I've updated this now.

I've tested the latest commit now builds using g++12 for hip backend within a docker locally. Assuming that there is no backend that is incompatible with being built by g++12, this should now be safe.

**Review by aelovikov-intel — APPROVED** [review]

**Comment by aelovikov-intel** [comment]
I think this needs to be split in two parts. First, fix the images but keep pre/post-commit running on old containers. Once merged and new images are verified locally we can switch the pre/post-commit tasks to use latest images once again.

**Comment by JackAKirk** [comment]
> I think this needs to be split in two parts. First, fix the images but keep pre/post-commit running on old containers. Once merged and new images are verified locally we can switch the pre/post-commit tasks to use latest images once again.

Makes sense to me. I've made this change.

**Comment by npmiller** [comment]
@intel/llvm-gatekeepers this is ready to merge

Note that the containers for pre/post commit are currently fixed to an older version, so this will re-generate the latest but cannot break testing.

**Pull request merged**